decayAmplitude/isobarDecayTopology.cc

```diff
@@ -543,126 +552,102 @@ isobarDecayTopology::getIsospinSymmetrization()
-			// we want to swap from the group vector and the index of the particle we
-			// want to swap it with from the permutations vector. The whole thing is
-			// applied to the map vector.
-			for(unsigned int j = 0; j < group.size(); ++j) {
-				map.at(group.at(j)) = group.at(permutations.at(j));
+			// Make sure we are not swapping two indistinguishable particles.
+			if(fsParts.at(j)->name() == fsParts.at(map.at(j))->name()) {
+				breaking = true;
+				break;
 			}
 
-			// Some first checks if the permutation makes sense.
-			bool breaking = false;
-			for(unsigned int j = 0; j < map.size(); ++j) {
-
-				if(j == map.at(j)) {
-					continue;
-				}
-
-				// Make sure we are not swapping two indistinguishable particles.
-				if(fsParts.at(j)->name() == fsParts.at(map.at(j))->name()) {
+			// Check if two particles from the same isobar are being swapped.
+			for(unsigned int k = 0; k < map.size(); ++k) {
+				if((map.at(k) == k) || (j == k)) continue;
+				if(fromVertex(fsParts.at(j)) == fromVertex(fsParts.at(k))) {
 					breaking = true;
 					break;
 				}
-
-				// Check if two particles from the same isobar are being swapped.
-				for(unsigned int k = 0; k < map.size(); ++k) {
-					if((map.at(k) == k) || (j == k)) continue;
-					if(fromVertex(fsParts.at(j)) == fromVertex(fsParts.at(k))) {
-						breaking = true;
-						break;
-					}
-				}
-				if(breaking) {
-					break;
-				}
 			}
-
 			if(breaking) {
-				continue;
+				break;
 			}
+		}
 
-			// Set the isospin of the final-state particles as given by the premutation.
-			for(unsigned int j = 0; j < map.size(); ++j) {
-				fsParts.at(j)->setIsospinProj(isospinProjs.at(map.at(j)));
-			}
-			calcIsobarCharges(false);
-
-			// Check for isospin consistency in all vertices.
-			for(unsigned int j = 0; j < _isobarVertices.size(); ++j) {
-				const particlePtr d1 = _isobarVertices.at(j)->daughter1();
-				const particlePtr d2 = _isobarVertices.at(j)->daughter2();
-				const particlePtr p  = _isobarVertices.at(j)->parent();
-				if(!spinStatesCanCouple(d1->isospin(), d1->isospinProj(),
-				                        d2->isospin(), d2->isospinProj(),
-				                        p->isospin(),  p->isospinProj())) {
-					breaking = true;
-				}
+		if(breaking) { // check next permutation
+			continue;
+		}
+
+		// Set the isospin of the final-state particles as given by the premutation.
+		for(unsigned int j = 0; j < map.size(); ++j) {
+			fsParts.at(j)->setIsospinProj(isospinProjs.at(map.at(j)));
+		}
+		calcIsobarCharges(false);
+
+		// Check for isospin consistency in all vertices.
+		for(unsigned int j = 0; j < _isobarVertices.size(); ++j) {
+			const particlePtr d1 = _isobarVertices.at(j)->daughter1();
+			const particlePtr d2 = _isobarVertices.at(j)->daughter2();
+			const particlePtr p  = _isobarVertices.at(j)->parent();
+			if(!spinStatesCanCouple(d1->isospin(), d1->isospinProj(),
+					d2->isospin(), d2->isospinProj(),
+					p->isospin(),  p->isospinProj())) {
```

Comment:
Please correct the indentation of the `if` clause, use tabs to get to the same indentation level as the surrounding code and the spaces to align the lines.

Also, please use `not` instead of `!`.